### DISCUSSION TITLE
problems with the installation

### DISCUSSION THREAD

**Discussion by shadow3108** [fuatakgun/eufy_security#848] [Q&A]
Sorry to set up a new conversation, but i´m struggling with the installation.
I´m new to Ha and just would like to test it a bit, befor making a final desicion regarding my home automation system.
therefore i´m not really familar with the technical details and issues from Ha. 
so far i´m qutie impressed how far i´m already, without deeper knogled.
now i would like to bring my eufy cams into Ha but i´m stucked somewhere in the setup process...

would you be so kind to support me, to get it running?

what i have done so far:

1. HACs integration of "Eufy Security"
2. installed the addon "eufy-security-ws"

looks like i put in some wrong values somewhere, because if i try to start the configuration of "Eufy Security", nothing happens.
![image](https://github.com/fuatakgun/eufy_security/assets/138705825/8982f1d1-6bcb-440d-98d5-983adf5b5cab)
the status won´t change?

I´m not sure if i did the correct entries in "eufy-security-ws"
![image](https://github.com/fuatakgun/eufy_security/assets/138705825/6a174f34-6ba7-49d6-9fba-3c38954c40d8)
what could be the issue?  "eufy-security-ws" is running ;-)

Thank you in advance

**Comment by bernhard4116** [comment]
I'm struggling with the exact same issue.. Help would be appreciated. Thank you

**Comment by tomschlde** [comment]
Hi all, same here.

The dialog does not come to an end.
The log says:
`Logger: frontend.js.latest.202307051
Source: components/system_log/__init__.py:257
First occurred: 11:14:13 (2 occurrences)
Last logged: 11:18:47

http://192.168.178.57:8123/frontend_latest/27811-wBwLJM4buZA.js:1:78234 Uncaught TypeError: Cannot read properties of null (reading 'removeEventListener')`

The integration stays in "initialize".

the reconfiguring dialog seems also broken, saving doesnt work:
![image](https://github.com/fuatakgun/eufy_security/assets/59179041/7fb07682-cd91-42e3-a471-076b96b76eda)

switching the checkbox "ffmepeg Protokolle" off and on again makes saving possible.

The integration has anaother log entry:

<Logger: homeassistant.config_entries
Source: custom_components/eufy_security/eufy_security_api/api_client.py:350
Integration: Eufy Security ([documentation](https://github.com/fuatakgun/eufy_security), [issues](https://github.com/fuatakgun/eufy_security/issues))
First occurred: 09:55:58 (2 occurrences)
Last logged: 11:19:28

Error setting up entry 127.0.0.1 for eufy_security
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 390, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/eufy_security/__init__.py", line 50, in async_setup_entry
    await coordinator.initialize()
  File "/config/custom_components/eufy_security/coordinator.py", line 39, in initialize
    await self._api.connect()
  File "/config/custom_components/eufy_security/eufy_security_api/api_client.py", line 73, in connect
    await self._set_products()
  File "/config/custom_components/eufy_security/eufy_security_api/api_client.py", line 100, in _set_products
    result = await self._start_listening()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/eufy_security/eufy_security_api/api_client.py", line 147, in _start_listening
    return await self._send_message_get_response(OutgoingMessage(OutgoingMessageType.start_listening))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/eufy_security/eufy_security_api/api_client.py", line 350, in _send_message_get_response
    return await future
           ^^^^^^^^^^^^
asyncio.exceptions.CancelledError>

**Reply by tomschlde** [reply]
more logs from anather try today after update of eufy_security_ws



'Logger: custom_components.eufy_security.config_flow
Source: custom_components/eufy_security/config_flow.py:89
Integration: Eufy Security ([documentation](https://github.com/fuatakgun/eufy_security), [issues](https://github.com/fuatakgun/eufy_security/issues))
First occurred: 10:51:05 (1 occurrences)
Last logged: 10:51:05

eufy_security Exception in login : - traceback: Traceback (most recent call last): File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection return await self._loop.create_connection(*args, **kwargs) # type: ignore[return-value] # noqa ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1085, in create_connection raise exceptions[0] File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1069, in create_connection sock = await self._connect_sock( ^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/asyncio/base_events.py", line 973, in _connect_sock await self.sock_connect(sock, address) File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 634, in sock_connect return await fut ^^^^^^^^^ File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 674, in _sock_connect_cb raise OSError(err, f'Connect call failed {address}') ConnectionRefusedError: [Errno 111] Connect call failed ('127.0.0.1', 3000) The above exception was the direct cause of the following exception: Traceback (most recent call last): File "/config/custom_components/eufy_security/eufy_security_api/web_socket_client.py", line 42, in connect self.socket = await self.session.ws_connect(f"ws://{self.host}:{self.port}", autoclose=False, autoping=True, heartbeat=60) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 779, in _ws_connect resp = await self.request( ^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 536, in _request conn = await self._connector.connect( ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 540, in connect proto = await self._create_connection(req, traces, timeout) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 901, in _create_connection _, proto = await self._create_direct_connection(req, traces, timeout) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1209, in _create_direct_connection raise last_exc File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1178, in _create_direct_connection transp, proto = await self._wrap_create_connection( ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection raise client_error(req.connection_key, exc) from exc aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host 127.0.0.1:3000 ssl:default [Connect call failed ('127.0.0.1', 3000)] The above exception was the direct cause of the following exception: Traceback (most recent call last): File "/config/custom_components/eufy_security/config_flow.py", line 113, in _test_credentials await api_client.ws_connect() File "/config/custom_components/eufy_security/eufy_security_api/api_client.py", line 67, in ws_connect await self._client.connect() File "/config/custom_components/eufy_security/eufy_security_api/web_socket_client.py", line 44, in connect raise WebSocketConnectionError() from exc custom_components.eufy_security.eufy_security_api.exceptions.WebSocketConnectionError'

**Reply by tomschlde** [reply]
I think, i will add an issue for that.

**Comment by shadow3108** [comment]
Hi guys,
sorry for my late reply, but I was on holiday.

unfortunately I can't really give you a hint how to solve the problem, but my system is running now.
Don't ask me what I did, from my point of view the same as before.

I think the cameras go into some kind of standby mode and can't be found.
only when the cameras are running and detect something can they be found. but that's just a feeling.
Maybe it is helpful?

**Reply by reimargithub** [reply]
Hi,
could you please copy your configuration settings in here to check if theres any mismatch ? I have the same problems as described above since 1-2 months and haven't made any changes.

**Reply by shadow3108** [reply]
From my point of view there is nothing special.


![image](https://github.com/fuatakgun/eufy_security/assets/138705825/556834fa-7a68-415d-9dec-b7ba91866a87)

**Comment by TobiMarn** [comment]
Hey,
I´ve a smiliar problem with eufy-security-ws. The interface worked very well till last week. I don´t what happend.

I´ve installed the Add-On again...but it is still not working.

I have these messages:

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
2023-08-17 07:59:40.681  INFO  Eufy Security server listening on host 0.0.0.0, port 3000 
2023-08-17 07:59:41.852  ERROR JSON parse error 
{
  data: '{"r',
  error: SyntaxError: Unexpected end of JSON input
      at JSON.parse (<anonymous>)
      at parseJSON (/usr/src/app/node_modules/eufy-security-client/build/utils.js:147:21)
      at Function.readValue (/usr/src/app/node_modules/eufy-security-client/build/http/parameter.js:50:63)
      at Camera.updateRawProperty (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:127:57)
      at /usr/src/app/node_modules/eufy-security-client/build/http/device.js:54:22
      at Array.forEach (<anonymous>)
      at Camera.update (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:53:35)
      at Camera.initializeState (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:29:14)
      at Camera.initialize (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:36:14)
      at /usr/src/app/node_modules/eufy-security-client/build/eufysecurity.js:638:32
}
2023-08-17 07:59:41.856  WARN Non-parsable parameter value received from eufy cloud. Will be ignored. 
{
  type: 2707,
  value: '{"r'
}
2023-08-17 07:59:41.857  ERROR JSON parse error 
{
  data: '{"r',
  error: SyntaxError: Unexpected end of JSON input
      at JSON.parse (<anonymous>)
      at parseJSON (/usr/src/app/node_modules/eufy-security-client/build/utils.js:147:21)
      at Function.readValue (/usr/src/app/node_modules/eufy-security-client/build/http/parameter.js:50:63)
      at Camera.updateRawProperty (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:127:57)
      at /usr/src/app/node_modules/eufy-security-client/build/http/device.js:54:22
      at Array.forEach (<anonymous>)
      at Camera.update (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:53:35)
      at Camera.initializeState (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:29:14)
      at Camera.initialize (/usr/src/app/node_modules/eufy-security-client/build/http/device.js:36:14)
      at /usr/src/app/node_modules/eufy-security-client/build/eufysecurity.js:638:32
}
2023-08-17 07:59:41.858  WARN Non-parsable parameter value received from eufy cloud. Will be ignored. 
{
  type: 2708,
  value: '{"r'
}
2023-08-17 07:59:43.593  INFO  Push notification connection closed 
2023-08-17 07:59:50.795  INFO  Connected to station T8010P23201213F1 on host 192.168.178.55 and port 16447 
2023-08-17 07:59:50.865  INFO  Push notification connection successfully established 
2023-08-17 07:59:51.055  INFO  Push notification connection successfully established 


Further necessary informations:
- I´m using a secondary account for HomeAssistant
- The two factor authentication is on

Thanks for your help/input.
